src/shaders/fill_extrusion_pattern.fragment.glsl

```diff
@@ -38,7 +43,18 @@ void main() {
     vec4 out_color = mix(color1, color2, u_fade);
 
     out_color = out_color * v_lighting;
-
+#ifdef FAUX_AO
+    float intensity = 1.0 - sqrt(1.0 - u_ao[0]);
+    float h = max(0.0, v_ao.z);
+    float h_floors = h / u_ao[1];
+    float y_shade = mix(1.0, 0.1, intensity * min(v_ao.y, 1.0));
```

Comment:
Same as above, we could move this `min` statement from fragment to vertex?

Reply:
Yes, the same as in https://github.com/mapbox/mapbox-gl-js/pull/11926#discussion_r892771397, it is interpolated from value greater than 1 (for fragments under ground).